[CS2113-W13-1] Your Mother’s Favorite CookBook (YMFC)

gskang-22 opened the pull request:
Cooks these days have more recipes than they know how to handle, and our product will help them store, retrieve and search through their recipes with ease. Prompts, tags and ingredients can be used to search a curated database, and recommend random recipes that closely match the criteria

Kobot7 commented on src/main/java/ymfc/YMFC.java:
```diff
@@ -0,0 +1,56 @@
+
+public class YMFC {
+    public static Logger logger = Logger.getLogger(YMFC.class.getName());
+    private static final String saveFilePath = "./data/recipes.txt";
```
Constant names must be all uppercase using underscore to separate words

Kobot7 commented on src/main/java/ymfc/YMFC.java:
```diff
@@ -0,0 +1,56 @@
+                    saidBye = true;
+                    logger.log(Level.INFO, "User said bye");
+                }
+            } catch (Exception e) {
```
Is the variable name "e" informative enough?

Kobot7 commented on src/main/java/ymfc/parser/Parser.java:
```diff
@@ -0,0 +1,172 @@
+
+    /**
+     * Parses user input commands and return a <code>Command</code> object
+     * @param commandString Input command as <code>String</code>
```
There should be punctuation after every parameter description

Kobot7 commented on src/main/java/ymfc/parser/Parser.java:
```diff
@@ -0,0 +1,172 @@
+ */
+public final class Parser {
+    // First word <command> before the first white space, then all the arguments <args>
+    // static final Pattern GENERIC_FORMAT = Pattern.compile("(?<command>\\S+)\\s+(?<args>.*)");
```
Do not leave code commented in your program. If unused, just remove it

Kobot7 commented on src/main/java/ymfc/parser/Parser.java:
```diff
@@ -0,0 +1,172 @@
+    private static SortCommand getSortCommand(String args) throws InvalidArgumentException {
+        final Pattern sortCommandFormat  =
+                Pattern.compile("(?<name>[sS]/[^/]+)");
+        args = args.trim();
```
Do not reuse formal parameters as local variables inside the method

Kobot7 commented on src/main/java/ymfc/parser/Parser.java:
```diff
@@ -0,0 +1,172 @@
+        } else {
+            return new AddRecipeCommand(new Recipe(name, ingreds, steps));
+        }
+    }
```
Try to avoid long methods. Perhaps you could extract some of the functionality into another method

Kobot7 commented on src/main/java/ymfc/parser/Parser.java:
```diff
@@ -0,0 +1,172 @@
+            return new AddRecipeCommand(new Recipe(name, ingreds, steps, timeTaken));
+        } else {
+            return new AddRecipeCommand(new Recipe(name, ingreds, steps));
+        }
```
Are these conditions really necessary?

Kobot7 commented on src/main/java/ymfc/recipelist/RecipeList.java:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    public boolean removeRecipeByName(String name) {
+        assert counter > 0 : "List should not be empty when deleting recipe";
```
Is this intended to be an assertion or exception?

Kobot7 commented on src/main/java/ymfc/storage/Storage.java:
```diff
@@ -0,0 +1,65 @@
+        if (isEmpty) {
+            return;
+        }
+        ListCommand c = new ListCommand();
```
Highly recommend using more informative variable names, "c" doesn't give much context on what the variable refers to

gskang-22 committed:
Add sequence UML for AddRecipe

gskang-22 committed:
Update command section for DG

gskang-22 committed:
Change isBye attribute in Command to non-static

gskang-22 committed:
Add UML sequence diagram for Command subclass AddRecipeCommand

gskang-22 committed:
Add Command section for DG

gskang-22 committed:
Add Command section for DG

gskang-22 committed:
Merge remote-tracking branch 'origin/branch-DG' into branch-DG

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/puml/AddRecipes.puml

gskang-22 committed:
Merge remote-tracking branch 'origin/branch-DG' into branch-DG

gskang-22 committed:
Solve merge conflicts

3CCLY committed:
Merge pull request #105 from gskang-22/branch-DG

Command part for DG

seanngja committed:
Fix visibility of attributes in Recipe class

Update attributes to be private instead of public

seanngja committed:
Merge pull request #106 from seanngja/A-Recipe

Fix visibility of attributes in Recipe class

gskang-22 committed:
Add and improve JavaDoc comments

gskang-22 committed:
Merge pull request #107 from gskang-22/branch-DG

Add and improve JavaDoc comments

KSanjith committed:
Add class diagram for r

KSanjith committed:
Add class diagram for ingredient classes

KSanjith committed:
Fix ingredient class diagram

KSanjith committed:
Rename ingredientsClassDiagram image

3CCLY committed:
Add UML diagram for edit command

3CCLY committed:
Update DG with UML diagram

seanngja committed:
Merge pull request #109 from 3CCLY/branch-add-uml

Add UML for EditCommand

Sukkaito committed:
docs: Update `Recipe` and `RecipeList` sections of DG

KSanjith committed:
Re-order ingredientsClassDiagram

KSanjith committed:
Add constructors to all class diagrams

KSanjith committed:
Merge pull request #108 from KSanjith/branch-ingredient-diagrams

Add ingredients class diagram

KSanjith committed:
Merge branch 'master' into docs/DG

KSanjith committed:
Merge pull request #110 from Sukkaito/docs/DG

docs: Update `Recipe` and `RecipeList` sections of DG

KSanjith committed:
Neaten the UG and DG

KSanjith committed:
Fix formatting in DG

Sukkaito committed:
Merge pull request #111 from KSanjith/branch-neaten-DGUG

Neaten the UG and DG

KSanjith committed:
Fix PDF page breaks

Atulteja commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+This class handles the addition, deletion and editing of recipes.
+
+Shown below is a class diagram detailing the interaction between `RecipeList` and `Recipe`:
+![RecipeClassDiagram.png](img/ClassDiagrams/recipesClassDiagram.png)
```
Need to go easy on the colours, maybe just use colours for the object lifeline

Atulteja commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![ArchitectureDiagram.png](img/ClassDiagrams/ArchitectureDiagram.png)
```
The class diagram is not supposed to have the user.

Atulteja replied:
The hard disc should not be there even outside the YMFC main,  it is understood that the storage will save the files to the hard drive

coraleaf0602 replied:
Could considering including labels with your association arrows in the class diagrams

Atulteja commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+
+[Sanjith](http://github.com/KSanjith/ip)
 
 ## Design & implementation
```
could seperate the design and implementation parts, under design u include th class diagrams and under implementation you have the sequence diagrams

Zhang-Zhitong commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+Shown below is a sequence diagram detailing how a new recipe is added using the AddRecipeCommand object,
+starting from parsing the user input to the addition of the recipe to the recipeList object,
+to finally printing the "recipe added" message to the User on the CLI:
+![AddRecipes.png](img/SequenceDiagrams/AddRecipesSequenceDiagram.png)
```
could the background be a less eye straining colour like white or green?

Zhang-Zhitong replied:
also, are the arrowheads filled or lined?

Zhang-Zhitong commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+Shown below is a sequence diagram detailing how an existing recipe is modified using the EditCommand object,
+starting from parsing the user input to finding the exiting recipe in the RecipeList object,
+to finally modifying and saving the recipe in the RecipeList:
+![EditCommand.png](img/SequenceDiagrams/EditRecipesSequenceDiagram.png)
```
could the return arrow be on the activation bar of :edit command instead of after it?

coraleaf0602 replied:
same issue as above - Could include the user to show how the sequence diagram is initialised - i.e. entering a certain input

Zhang-Zhitong commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+
+Shown below is a sequence diagram detailing how the Parser class interprets the user's CLI input
+and returns the appropriate command object containing the user's input parameters.
+![Parser.parseCommand.png](img/SequenceDiagrams/Parser.parseCommand.png)
```
looks good to me.

kestryix replied:
Would it be possible to standardise the diagram colours as the diagrams have different colour schemes and are using very bright colours that make it hard to read?

kestryix replied:
Also, would it be possible to break the sequence diagram down into smaller sequence diagrams to make it clearer and more distinct for each command?

coraleaf0602 replied:
Looks good but could consider splitting the Parser class to a smaller component if its sequence diagram is so big

Zhang-Zhitong commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+  - The following Sequence diagram shows how the saveRecipes() method in the Storage class
+  saves all the created recipes into the .txt file
+
+![saveRecipes.png](img/SequenceDiagrams/saveRecipesSequenceDiagram.png)
```
could the return arrow on the :storage activation bar be aligned?

kestryix commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+  - `execute()`: An abstract method that must be implemented by all subclasses
+    - Defines the primary action performed by the command (E.g. add new recipe to `recipeList` for `AddRecipeCommand`)
+
+Shown below is a class diagram detailing the interaction between the parent `Command` class and it's child classes:
+![commandClasses.png](img/ClassDiagrams/commandsClassDiagram.png)
```
Would it be possible to break this class diagram down or format it so that it is more easily viewed?

kestryix commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+Shown below is a sequence diagram detailing how a new recipe is added using the AddRecipeCommand object,
+starting from parsing the user input to the addition of the recipe to the recipeList object,
+to finally printing the "recipe added" message to the User on the CLI:
+![AddRecipes.png](img/SequenceDiagrams/AddRecipesSequenceDiagram.png)
```
Would it be possible to make the colours more neutral and easily viewable? Also, for the notations, perhaps you could double check them to ensure that they follow the necessary guidelines?

Yisoong replied:
Last ref is out of bounds of the diagram

coraleaf0602 replied:
Could include the user to show how the sequence diagram is initialised - i.e. entering a certain input

Yisoong commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+  1. Update the `parseCommand()` method in `Parser` to recognise and instantiate the new command
+
+#### Example of `Command` Child Classes
+
```
Might want to choose colours that are less bright, or standardise colours across the different diagrams

Yisoong commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+
+[Sanjith](http://github.com/KSanjith/ip)
 
 ## Design & implementation
```
The DG might benefit from having a contents page

coraleaf0602 replied:
Agreed. Could add a contents section with each of the headings having their own corresponding link

Yisoong commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+to finally printing the "recipe added" message to the User on the CLI:
+![AddRecipes.png](img/SequenceDiagrams/AddRecipesSequenceDiagram.png)
+
+2. `EditCommand` Class
```
It would be good to show the user in the sequence diagrams to know where the commands are coming from

kestryix commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+to finally modifying and saving the recipe in the RecipeList:
+![EditCommand.png](img/SequenceDiagrams/EditRecipesSequenceDiagram.png)
+
+### Parser Class
```
For the parser and storage classes, would it be possible to elaborate more and standardise the formats to be similar to what was done for the command class?

kestryix reviewed (commented):
Generally okay, could be possible to standardise the formats so that the DG is more cohesive. Colours and diagrams can also be standardise to improve reading experience.

Yisoong commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+### How the architecture components interact with each other
+The sequence diagram below shows how the components interact with each other when the application is first launched.
+
+![ApplicationStartUp.png](img/SequenceDiagrams/ApplicationStartUp.png)
```
Might be good to have a return arrow back to the user

coraleaf0602 replied:
Could consider adding labels above the return arrows back to the user

Yisoong reviewed (commented):
Good work!

gskang-22 committed:
Add to UG

gskang-22 committed:
Add to UG

gskang-22 committed:
Merge pull request #114 from gskang-22/branch-DG&UG

Add to UG

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Certain aspects of YMFC's code was inspired from the following member(s) IPs:
```
I like how you referenced team member's code if needed!

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,207 @@
+The command class deals with all the possible commands accepted by YMFC, with each command representing a specific 
+recognised user input (E.g. ListCommand representing the command to list all the recipes in the recipeList).
 
+#### 1. The `Command` base class
```
I find this section a bit difficult to read as there's too much information squished together, could consider simplifying the class diagram or spacing out the sections

Sukkaito committed:
bugfix: Custom message for no recipe or ingredient found

KSanjith committed:
Write PPP for KSanjith

Sukkaito committed:
bugfix: Disallow duplicate value in `options` field of `find` command

3CCLY committed:
Merge pull request #119 from KSanjith/branch-PPP

Write PPP for KSanjith

3CCLY committed:
Merge pull request #116 from Sukkaito/bugs/113

bugfix: Custom message for no recipe or ingredient found

3CCLY committed:
Create draft PPP

KSanjith committed:
Merge pull request #121 from 3CCLY/branch-PPP-draft

Create draft PPP

KSanjith committed:
Merge pull request #112 from KSanjith/branch-PDF-formatting

Fix PDF formatting

KSanjith committed:
Merge pull request #120 from Sukkaito/bugs/73

bugfix: Disallow duplicate value in `options` field of `find` command

Sukkaito committed:
docs: Sukkaito's PPP draft

seanngja committed:
Create draft PPP

seanngja committed:
Add findI instructions to help command

seanngja committed:
Update User Guide with findI command

seanngja committed:
Update PPP with updates in User Guide

seanngja committed:
Update I/O redirection test expected inputs for findI

seanngja committed:
Fix grammar in User Guide

KSanjith committed:
Merge pull request #124 from seanngja/branch-UserGuide

Update User Guide with findI command

KSanjith committed:
Merge pull request #123 from seanngja/branch-PPP

seanngja's draft PPP

gskang-22 committed:
Add PPP for gskang to describe contribution to the project

gskang-22 committed:
Add PPP for gskang to describe contribution to the project

gskang-22 committed:
Merge remote-tracking branch 'origin/branch-DG&UG' into branch-DG&UG

Sukkaito committed:
Merge pull request #125 from gskang-22/branch-DG&UG

Add PPP for gskang

Sukkaito committed:
docs: Draft PPP for Sukkaito

Sukkaito committed:
Merge branch 'AY2425S1-CS2113-W13-1:master' into docs/PPP

Sukkaito committed:
typo: fix ParameterizedTest hyperlink

seanngja committed:
Merge pull request #122 from Sukkaito/docs/PPP

Sukkaito's PPP draft

Sukkaito committed:
bugfix: show correct logs for missing savefile

Sukkaito committed:
cleanup: remove unused imports

KSanjith committed:
Merge pull request #127 from Sukkaito/bugs/115

bugfix: Show correct logs for missing savefile

KSanjith committed:
Correct issues in DG pointed out by TA

gskang-22 committed:
Add Junit test IngredientListTest for IngredientList

gskang-22 committed:
Replace counter attribute in RecipeList with .size() function of ArrayList

gskang-22 committed:
Implement DeleteIngredient feature for "deleteI n/ingredient"

gskang-22 committed:
Removed extra methods from IngredientList and RecipeList
- Removed removeIngredient and removeRecipe methods and updated their corresponding test cases

gskang-22 committed:
Remove unused import

gskang-22 committed:
Update imports to fix checkstyleerror

gskang-22 committed:
Update imports to fix checkstyleerror

KSanjith committed:
Fix commandsShortClassDiagram

gskang-22 committed:
Fix issue #72: bug:false behavior of add command

gskang-22 committed:
Fix checkstyleerror

Sukkaito committed:
Merge pull request #131 from gskang-22/branch-debug

Close Issue #72 and #76

Sukkaito committed:
bugfix: trailing/leading whitespaces are considered part of the elements

Changes:
- Accept `String` arguments with trailing/leading spaces, but will be trimmed before adding to the database.
- Adjust `timeTaken` argument pattern and `getTimeTakenInteger()` to accept trailing/leading spaces (for consistency).

3CCLY committed:
Implement random recommendation command

3CCLY committed:
Add recommend command based on available ingredients

3CCLY committed:
Show percentage of available ingredients for recommend command

3CCLY committed:
Show missing ingredients with recommend command

3CCLY committed:
Package recommended recipe parameters into its own class

3CCLY committed:
Remove unused methods

3CCLY committed:
Merge pull request #128 from KSanjith/branch-fix-DG

Correct issues in DG pointed out by TA

seanngja committed:
Update Recipe's ingredient list to be Ingredient

ArrayList of string changed to ArrayList of Ingredient

seanngja committed:
Update Recipe class equals method

KSanjith committed:
Merge pull request #135 from seanngja/A-Recipe

Update Recipe class ingredients list type to ArrayList of Ingredients

3CCLY committed:
Merge branch 'master' into branch-recommend-command

3CCLY committed:
Update recommend command to use new recipe class

3CCLY committed:
Remove redundant IngredientList method

gskang-22 committed:
Merge pull request #134 from 3CCLY/branch-recommend-command

Implement recommend function to application

gskang-22 committed:
Add to command class portion of the DG

3CCLY committed:
Merge pull request #136 from gskang-22/branch-Tests

Addition to DG

Sukkaito committed:
Merge pull request #137 from Sukkaito/bugs/129

bugfix: trailing/leading whitespaces are considered part of the elements

Sukkaito committed:
typofix: Last item of ToC doesn't reference to anywhere (#139)

* typo: Fix command summary hyperlink not working for ToC in Github Page

Sukkaito committed:
typo: Fix broken hyperlinks in ToC

Sukkaito committed:
Merge pull request #140 from Sukkaito/docs/UG

typofix: Fix broken hyperlinks in ToC

3CCLY committed:
Add tests for RecommendedRecipe and SortByPercentMatch classes

KSanjith committed:
Allow user to sort ingredients by name
Prevent duplicate named ingredients from being added

Sukkaito committed:
test: Increase test coverage of `Parser.java`

Changes:
- Add various tests.
- Enable past tests that were disable before bug fixing.
- Add `validateStepNumbers()` for `edit` command.

KSanjith committed:
Update ksanjith ppp

Sukkaito committed:
test: Increase test coverage for `AddIngredientCommand.java`

Sukkaito committed:
test: Increase test coverage for `DeleteIngredientCommand.java`

3CCLY committed:
Add tests for RandomCommand and RecommendCommand classes

3CCLY committed:
Merge pull request #142 from KSanjith/branch-sort-ingreds

Create Sort Ingredients Command

3CCLY committed:
Merge branch 'master' into branch-new-test

Sukkaito committed:
hotfix: follow CheckStyle of coding conventions

3CCLY committed:
Update PPP

3CCLY committed:
Update DG with recently added recommend and random command

3CCLY committed:
Merge pull request #143 from Sukkaito/feature/junit-test

Increase test coverage

3CCLY committed:
Merge pull request #144 from 3CCLY/branch-new-test

Add JUnit Test for recent commands

Sukkaito committed:
test: Further push test coverage

Sukkaito committed:
hotfix: follow coding convention

Sukkaito committed:
Merge pull request #145 from Sukkaito/feature/junit-test

Push test coverage further

gskang-22 committed:
Refactor names of classes to follow naming convention
- AddRecipeCommand -> AddCommand
- ListIngredientsCommand -> ListIngredientCommand

gskang-22 committed:
Merge pull request #146 from gskang-22/branch-DG

Refactor names of classes to follow naming convention

gskang-22 committed:
Add test for RecommendCommand when IngredientList is empty

gskang-22 committed:
Add tests for Commands and debugged Ui
- AddCommandTest -> test for when trying to add duplicate recipe
- AddCommandTest -> test for when trying to add duplicate ingredient
- Fixed issue where Ui error message would contain "recipe" when trying to add duplicate Ingredient

gskang-22 committed:
Add test for Ui PrintErrorMessage() method

gskang-22 committed:
Fix CheckstyleError

seanngja committed:
Rename findIngredCommand to findIngredientCommand

KSanjith committed:
Merge pull request #147 from gskang-22/branch-DG

Added more tests

KSanjith committed:
Merge pull request #148 from seanngja/branch-findIngreds

Rename findIngredCommand to findIngredientCommand

seanngja committed:
Add description of findIngredientCommand in DG

seanngja committed:
Add user stories under v2.0 to DG

3CCLY committed:
Merge pull request #151 from seanngja/branch-DG

Add description of findIngredientCommand in DG

Sukkaito committed:
docs: Update `FindCommand` class description of DG

Sukkaito committed:
Merge pull request #152 from Sukkaito/docs/DG

docs: Update `FindCommand` class description of DG

Sukkaito committed:
typo: attempt to fix random typos

Sukkaito committed:
Merge branch 'AY2425S1-CS2113-W13-1:master' into docs/DG

KSanjith committed:
Standardise and update documentations

KSanjith committed:
Merge branch 'master' into branch-standardise-DG

3CCLY committed:
Update UG

seanngja committed:
Merge pull request #153 from KSanjith/branch-standardise-DG

Standardise and update documentations

3CCLY committed:
Resolve merge conflict

KSanjith committed:
Merge pull request #154 from 3CCLY/branch-final-UG

Update UG

Sukkaito committed:
Merge branch 'AY2425S1-CS2113-W13-1:master' into docs/DG

Sukkaito committed:
docs: Attempt to fix inappropriate headers

Sukkaito committed:
docs: fix incorect indent

KSanjith committed:
Add asserts

seanngja committed:
Merge pull request #155 from KSanjith/branch-add-an-assert

Add asserts

Sukkaito committed:
Merge branch 'AY2425S1-CS2113-W13-1:master' into docs/DG

seanngja committed:
Fix bug in FindCommand

seanngja committed:
Merge pull request #156 from Sukkaito/docs/DG

docs: Fix formatting of the DG

Sukkaito committed:
docs: terminate lifelines in sequence diagrams

seanngja committed:
Trim command string in parserCommand method

seanngja committed:
Merge pull request #159 from Sukkaito/docs/DG

docs: terminate lifelines in sequence diagrams

KSanjith committed:
Merge pull request #158 from seanngja/branch-parser

Fix bug in FindCommand

Sukkaito committed:
docs: fix UG typo

Sukkaito committed:
docs: Attempt to fix UG break lines

Sukkaito committed:
docs: Attempt to improve DG ability to pdf printing

Sukkaito committed:
Merge pull request #160 from Sukkaito/master

Add line breaks to UG to generate pdf file

Sukkaito committed:
docs: Update Home Page

Sukkaito committed:
Merge branch 'AY2425S1-CS2113-W13-1:master' into master

Sukkaito committed:
Merge pull request #161 from Sukkaito/master

Update Home Page

seanngja committed:
Add names to objects in start up sequence diagram

seanngja committed:
Update start up sequence diagram in DG

seanngja committed:
Fix leading spaces bug in delete commands

delete and deleteI now will match the recipes and ingredients to be
deleted if leading spaces are inputted by the user

seanngja committed:
Update IO redirection test for deleteI command

KSanjith committed:
Merge pull request #214 from seanngja/branch-bugfix

Fix leading spaces bug in delete commands, fix incorrect UI when deleteI command is executed

KSanjith committed:
Merge pull request #213 from seanngja/branch-DG

Add names to objects in start up sequence diagram

KSanjith committed:
Enhance saving and loading

KSanjith committed:
Fix checkstyle violations

3CCLY committed:
Merge pull request #215 from KSanjith/branch-enhance-saving

Enhance saving and loading

3CCLY committed:
Modify edit command pattern

3CCLY committed:
Improve edit command to edit recipe parameters independently

KSanjith committed:
Create save files upon launch of app

3CCLY committed:
Prevent duplicate recipe names through edit command

3CCLY committed:
Fix tests for edit command

3CCLY committed:
Allow renaming of existing recipe to original name but different case

3CCLY committed:
Update UG

3CCLY committed:
Update UG

3CCLY committed:
Add message for edit command with no optional parameters

Sukkaito committed:
Merge pull request #216 from 3CCLY/branch-fix-edit

Improve edit command functionality

KSanjith committed:
Undo indents in certain exception messages

Sukkaito committed:
Merge pull request #217 from KSanjith/branch-enhance-saving

Create save files upon launch of app

KSanjith committed:
Updates storage related documentations

KSanjith committed:
Rename save files to cookbook.txt & inventory.txt

KSanjith committed:
Update the help ui and command summaries

3CCLY committed:
Merge pull request #219 from KSanjith/branch-update-storage-DG

Updates storage related documentations

Sukkaito committed:
docs: Fix `AddRecipes.puml` diagram

Sukkaito committed:
docs: `:AddCommand` lifeline now extends a bit further before terminating

gskang-22 committed:
Update Parser to deny steps in wrong order when adding recipes

gskang-22 committed:
Update Parser to deny steps in wrong order when adding recipes

gskang-22 committed:
fix checkstyleerror

gskang-22 committed:
fix checkstyleerror

Sukkaito committed:
docs: Change `:Recipe` (the object name) to `recipe:Recipe`

gskang-22 committed:
Add table of contents for DG

Sukkaito committed:
bugfix: Fix `FindCommand` isn't case-insensitive

Changes:
- `FindCommand` is now case-insensitive
- Fix the boolean operators to become short-circuit (terminates asap)

gskang-22 committed:
Add more user stories to DG

3CCLY committed:
Merge pull request #224 from gskang-22/branch-DG

DG update

3CCLY committed:
Merge pull request #222 from gskang-22/branch-debug

Debug: fix parser validateStepNumbers()

gskang-22 committed:
Deny addition of recipes with duplicate ingredients

gskang-22 committed:
Fix checkstyleerror

gskang-22 committed:
Merge remote-tracking branch 'origin/branch-debug' into branch-debug

# Conflicts:
#	src/main/java/ymfc/parser/Parser.java

gskang-22 committed:
Merge branch 'master' into branch-debug

gskang-22 committed:
Fix checkstyleerror

gskang-22 committed:
Merge remote-tracking branch 'origin/branch-debug' into branch-debug

# Conflicts:
#	src/main/java/ymfc/parser/Parser.java

gskang-22 committed:
Update to UG and PPP

gskang-22 committed:
Merge pull request #226 from gskang-22/branch-DG&UG

Update to UG and PPP

3CCLY committed:
Make recommend command ignore case for ingredients

3CCLY committed:
Merge pull request #225 from gskang-22/branch-debug

Debug: fix issue of duplicate ingredients in recipe

3CCLY committed:
Merge branch 'master' into branch-update-commands

3CCLY committed:
Fix duplicate ingredients for edit command

3CCLY committed:
Fix ghost code

Sukkaito committed:
docs: Add notes on command with no params to UG

3CCLY committed:
Update edit command sequence diagram

3CCLY committed:
Merge pull request #228 from Sukkaito/bugs/184

Add notes on command with no parameters to UG

Sukkaito committed:
docs: Add restriction for parameters order in UG

Sukkaito committed:
enhance: Add range for `TIME` parameter of `add` command

Changes:
- TIME is a positive integer between 1 and 65535, both end included.
- Show custom messages when receive invalid TIME input.

3CCLY committed:
Merge pull request #223 from Sukkaito/bugs/181

Fix `FindCommand` isn't case-insensitive

gskang-22 committed:
Update file name of gskang's PPP

gskang-22 committed:
Merge pull request #232 from gskang-22/branch-DG

Update file name of gskang's PPP

Sukkaito committed:
Merge branch 'AY2425S1-CS2113-W13-1:master' into docs/DG

Sukkaito committed:
Merge pull request #221 from Sukkaito/docs/DG

Fix some diagram-related minor and critical details of `AddRecipe.puml`

Sukkaito committed:
docs: Add missing `AddRecipesSequenceDiagram.png` after merging

Sukkaito committed:
Merge pull request #233 from Sukkaito/docs/DG

Add missing `AddRecipesSequenceDiagram.png` after merging #221

Sukkaito committed:
Merge branch 'AY2425S1-CS2113-W13-1:master' into bugs/small-fixes

Sukkaito committed:
enhance: Add range for `TIME` parameter of `edit` command

seanngja committed:
Merge pull request #230 from Sukkaito/bugs/small-fixes

Some random enhancements mentioned in the issue list

3CCLY committed:
Fix edit command sequence diagram

3CCLY committed:
Fix authorship for gskang-22

3CCLY committed:
Update PPP

3CCLY committed:
Add authorship to documentation

seanngja committed:
Merge pull request #227 from 3CCLY/branch-update-commands

Make recommend command ignore case for ingredients

seanngja committed:
Update PPP, claim authorship of lines in DG, UG

Sukkaito committed:
Merge pull request #234 from seanngja/branch-PPP

Update seanngja's PPP, claim authorship of lines in DG, UG

seanngja committed:
Add hyperlinks to PPP

seanngja committed:
Merge pull request #235 from seanngja/branch-PPP

Add hyperlinks to seanngja PPP

KSanjith committed:
Add authorship

KSanjith committed:
Add hyperlink to exception section in DG

KSanjith committed:
Fix grammer

Sukkaito committed:
Merge pull request #236 from KSanjith/branch-authorship

Add KSanjith authorship to DG and UG

Sukkaito committed:
docs: Fix last item of ToC doesn't point to anywhere in DG

KSanjith committed:
Add page breaks in PPP

KSanjith committed:
Merge pull request #237 from KSanjith/branch-authorship

Add page breaks in PPP

3CCLY committed:
Finalise PPP

3CCLY committed:
Merge pull request #240 from 3CCLY/branch-final-PPP

Finalise PPP

Sukkaito committed:
docs: Remove exceed lifeline and minor edits for `EditRecipeSequenceDiagram.png`

Changes:
- Manually remove redundant part of lifelines
- Reposition return arrow of `command:EditCommand` in `EditRecipeSequenceDiagram.png`

gskang-22 committed:
Update PPP and DG

gskang-22 committed:
Update DG

gskang-22 committed:
Merge pull request #241 from gskang-22/branch-DG

Update PPP and DG

Sukkaito committed:
docs: Format DG for nicer view

Sukkaito committed:
Merge branch 'master' into docs/DG

Sukkaito committed:
docs: Format UG to support PDF printing

Sukkaito committed:
Merge pull request #242 from Sukkaito/docs/DG

Finalize DG

Sukkaito committed:
Merge pull request #243 from Sukkaito/docs/UG

Format UG to support PDF printing

Sukkaito committed:
docs: Claiming authorship UG, DG and finalizing PPP

Sukkaito committed:
Merge pull request #244 from Sukkaito/master

Claiming authorship UG, DG and finalizing PPP

seanngja committed:
Fix language error in DG

Sukkaito committed:
Merge pull request #245 from seanngja/branch-DG

Fix language error in DG
